src/hash.rs

```diff
@@ -59,8 +60,16 @@ impl StarkFelt {
         Err(StarknetApiError::OutOfRange { string: hex_str_from_bytes::<32, true>(bytes) })
     }
 
-    /// Storage efficient serialization for field elements.
-    pub fn serialize(&self, res: &mut impl std::io::Write) -> Result<(), Error> {
+    /// Returns a new unckecked [`StarkFelt`].
+    ///
+    /// # Safety
+    /// The most significant byte nible of the array must be 0 (`byte[0] < 0x10`).
+    /// Otherwise the resulting [`StarkFelt`] will end up containing an out of range value.
+    pub const unsafe fn new_unchecked(bytes: [u8; 32]) -> Self {
```

Comment:
Why do we have unsafe code here ?

Reply:
but I get you point, it's not `unsafe` on memory management point of view

Reply:
Treated at compile time so it doesn't lead to undefined behaviors at runtime so it's fine :heavy_check_mark: